test/controllers/actual_db_schema/migrations_controller_test.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+require_relative "../../test_helper"
+require_relative "../../../app/controllers/actual_db_schema/migrations_controller"
+
+module ActualDbSchema
+  class MigrationsControllerTest < ActionController::TestCase
+    def setup
+      @utils = TestUtils.new
+      @app = Rails.application
+      routes_setup
+      Rails.logger = Logger.new($stdout)
+      ActionController::Base.view_paths = [File.expand_path("../../../app/views/", __dir__)]
+      active_record_setup
+      @utils.cleanup(TestingState.db_config)
+      @utils.prepare_phantom_migrations(TestingState.db_config)
+    end
+
+    def routes_setup
+      @routes = @app.routes
+      Rails.application.routes.draw do
```

Comment:
Still, why we should init routes here and why they are not initiated during engine loading? Maybe the problem is that the engine is not being loaded in tests? Then we should try to load it in tests?

These lines signify that the engine is not being loaded in tests and those lines are not covered by tests. And this is bad as the tests are fragile.